### DISCUSSION TITLE
Detection using yolov8

### DISCUSSION THREAD

**Discussion by CHIRU98** [ultralytics/ultralytics#4794] [Q&A]
"Hello! I hope you're doing well. I've attached a sample screenshot with below link, and based on that screenshot, I'm looking to get the results in JSON format. The expected JSON structure is as follows:

https://drive.google.com/file/d/19KHUchuFa6zhHevaP0oxZebQhk0NeARl/view?usp=sharing

For Group1:

First Row: {"name": "First Name", "Type": "Input"}
Second Row: {"Day": "DropDown", "Month": "DropDown", "Year": "DropDown"}

For Group2:
Third Row: {"Id": "", "Social Security Number": "", "9 digit Number": ""}
Could you please help me convert the screenshot data into this JSON format?"

**Comment by fabian57fabian** [comment]
YoloV8 has multiple pretrained models for following tasks:

- Classification
- Object detection
- Segmentation
- Tracking
- Pose estimation

None of these extracts text from image.

In order to deal with that, if your input has standard size, you could crop your interesting parts from the image and extract text with python library "tesseract".

Here is a python example code:

1. First, install tesseract (on ubuntu `sudo apt install tesseract-ocr`, on windows install manually)
2. Then install some libraries `python3 -m pip install pillow pytesseract`
3. Finally run following code (assuming your image is in current directory):

```python3
from PIL import Image
import pytesseract

# Open image
filename="03-RU-Account-Info.png"
img = Image.open(filename)
display(img)

# Crop a part
left, top, right, buttom = 50, 70, 50+300, 70+40
img_fullname = img.crop((left, top, right, buttom))
display(img_fullname)

full_name = pytesseract.image_to_string(img_fullname)
print(full_name)

```

This will output **Pavel Zeifart**

**Reply by glenn-jocher** [reply]
@fabian57fabian hello!

Thank you for reaching out. It seems you're looking to perform Optical Character Recognition (OCR) to extract text from an image and convert it into a JSON format. While YOLOv8 is a powerful tool for computer vision tasks such as object detection, segmentation, and tracking, it does not natively perform OCR or structure data into JSON.

For your task, you would typically use an OCR library to extract text from the image, and then write a custom script to format the extracted text into the JSON structure you've described. Libraries like Tesseract can be used for OCR, and you can then use Python to parse the text and create the JSON.

Once you have the text, you would need to identify the patterns or positions of the text corresponding to each JSON field and map them accordingly. This process is specific to the layout and content of your images, so you'll need to tailor your script to handle the structure of your data.

If you have further questions on YOLOv8's capabilities, feel free to check out our documentation at https://docs.ultralytics.com. Good luck with your project! 😊
